oak/src/oak/server.cpp

```diff
@@ -175,75 +112,207 @@ class Request : public cloe::Request {
 
  public:
   /**
-   * Create a new Request from ServerImpl.
-   *
-   * - This should not panic.
+   * Creates a new Request from a http_connection
    */
-  explicit Request(const ServerImpl::request& req, const ServerImpl::connection_ptr& conn)
-      : req_(req)
-      , conn_(conn)
-      , endpoint_(Muxer<Handler>::normalize(req.destination))
-      , queries_(parse_queries(req.destination)) {
-    from_string(req.method, method_);
-    if (method_ == RequestMethod::POST) {
-      body_ = DataReceiver<ServerImpl>(req, conn).read_all();
-    }
-  }
+  explicit Request(const http_connection& connection);
 
   RequestMethod method() const override { return method_; }
   ContentType type() const override { return ContentType::UNKNOWN; }
   const std::string& body() const override { return body_; }
-  const std::string& uri() const override { return req_.destination; }
+  const std::string& uri() const override { return uri_; }
   const std::string& endpoint() const override { return endpoint_; }
   const std::map<std::string, std::string>& query_map() const override { return queries_; }
 };
 
-ServerImplHandler::ServerImplHandler() {
-  muxer.set_default([this](const cloe::Request& q, cloe::Response& r) {
-    ::oak::log()->debug("404 {}", q.endpoint());
-    r.not_found(Json{
-        {"error", "cannot find handler"},
-        {"endpoints", Json(this->muxer.routes())},
+/**
+ * @brief Represents the http-connection
+ */
+class http_connection : public std::enable_shared_from_this<http_connection> {
+ public:
+  http_connection(ip::tcp::socket socket, Muxer<Handler>& muxer)
+      : socket_(std::move(socket))
+#if (BOOST_VERSION / 100000 == 1)
+#if (BOOST_VERSION / 100 % 1000 > 69)
+      , deadline_(socket_.get_executor())
+#else
+      , deadline_(socket.get_io_service())
+#endif
+#else
+#error unexpected boost version
+#endif
+      , muxer_(muxer) {
+    deadline_.expires_after(std::chrono::seconds(60));
+  }
+
+  void start() {
+    read_request();
+    check_deadline();
+  }
+
+  /**
+   * @brief Returns the boost::beast request
+   *
+   * @return const request_type& The boost::beast request
+   */
+  inline const request_type& getRequest() const { return request_; }
+
+ private:
+  /**
+   * @brief The socket for the currently connected client.
+   */
```

Comment:
We don't need to use big `/** */` blocks for one-liners, `///` works there too, but I have always avoided to use Doxygen comments on private variables anyway. Leave the comments there that are helpful.